hada-manager/src/common/textFunctions.ts

```diff
@@ -0,0 +1,98 @@
+const EngToHeDict: Record<string, string> = {
+    a: 'ש',
+    b: 'נ',
+    c: 'ב',
+    d: 'ג',
+    e: 'ק',
+    f: 'כ',
+    g: 'ע',
+    h: 'י',
+    i: 'ן',
+    j: 'ח',
+    k: 'ל',
+    l: 'ך',
+    m: 'צ',
+    n: 'מ',
+    o: 'ם',
+    p: 'פ',
+    q: '/',
+    r: 'ר',
+    s: 'ד',
+    t: 'א',
+    u: 'ו',
+    v: 'ה',
+    w: '\'',
+    x: 'ס',
+    y: 'ט',
+    z: 'ז',
+    A: 'ש',
+    B: 'נ',
+    C: 'ב',
+    D: 'ג',
+    E: 'ק',
+    F: 'כ',
+    G: 'ע',
+    H: 'י',
+    I: 'ן',
+    J: 'ח',
+    K: 'ל',
+    L: 'ך',
+    M: 'צ',
+    N: 'מ',
+    O: 'ם',
+    P: 'פ',
+    Q: '/',
+    R: 'ר',
+    S: 'ד',
+    T: 'א',
+    U: 'ו',
+    V: 'ה',
+    W: '\'',
+    X: 'ס',
+    Y: 'ט',
+    Z: 'ז'
+};
+
+const HeToEngDict : Record<string,string> = {
+    'ש' : 'a',
+    'נ' : 'b',
+    'ב' : 'c',
+    'ג' : 'd',
+    'ק' : 'e',
+    'כ' : 'f',
+    'ע' : 'g',
+    'י' : 'h',
+    'ן' : 'i',
+    'ח' : 'j',
+    'ל' : 'k',
+    'ך' : 'l',
+    'צ' : 'm',
+    'מ' : 'n',
+    'ם' : 'o',
+    'פ' : 'p',
+    '/' : 'q',
+    'ר' : 'r',
+    'ד' : 's',
+    'א' : 't',
+    'ו' : 'u',
+    'ה' : 'v',
+    '\'' : 'w',
+    'ס' : 'x',
+    'ט' : 'y',
+    'ז' : 'z'
+}
```

Comment:
There is no need to specify this map again, when you can reverse the English to Hebrew map:
(You can read about the array reduce function so you understand it better.)


```suggestion
const HeToEngDict: Record<string, string> = Object.entries(EngToHeDict).reduce((acc, [eng, he]) => {
  if (!acc[he]) {
    acc[he] = eng;
  }
  return acc;
}, {} as Record<string, string>);
```

Reply:
🤯